### PR TITLE
feat: use metadata schema

### PR DESCRIPTION
**Pull request by Xillians**
see #623 for more information

this draft is to implement the meta data component in svelte

### CODE REVIEW

File: src/components/metadata.svelte [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
   import { urlFor } from '../data/sanity-client-browser';
   import type { Picture } from 'vite-imagetools';
+  import type { Metadata } from '../data/types';
 
-  export let title: string | undefined = undefined;
-  export let description: string | undefined = undefined;
-  export let data: MetaTagsProps = {};
-  export let seo: Seo | null = null;
+  export let data: Metadata = {
+    title: null,
```
Comment (braaar) [inline src/components/metadata.svelte:9]: This field is currently not defined in the schema in sanity, do we want to add it there, perhaps?

**Review by braaar — COMMENTED** [review]
This looks good, but I would consider making a separate component so that we can swap out the old one gradually

File: src/components/metadata.svelte [hunk 2]
```diff
@@ -1,20 +1,33 @@
 <script lang="ts">
-  import { MetaTags, type MetaTagsProps } from 'svelte-meta-tags';
-  import type { Seo } from '../data/types';
+  import { MetaTags } from 'svelte-meta-tags';
   import dedent from 'dedent';
   import { urlFor } from '../data/sanity-client-browser';
   import type { Picture } from 'vite-imagetools';
+  import type { Metadata } from '../data/types';
```
Comment (simenandre) [inline src/components/metadata.svelte]:
```suggestion
   export let title = string | undefined = undefined;
   export let searchEngine = SearchEngineMetadata | undefined = undefined;   
```

Can we do this instead?

File: src/data/types.ts [hunk 3]
```diff
@@ -14,3 +14,46 @@ export const seoModel = z.object({
...
+            alt: z.string().nullable()
+          })
+          .nullable()
+      )
+      .describe(
+        'The ideal size is 1200x630px. When you are providing multiple images, the first image will be used as the main image. Some social media platforms will allow the user to choose which image to use.'
+      )
+  })
+});
+export type Metadata = z.infer<typeof metadataModel>;
```
Comment (simenandre) [inline src/data/types.ts:59]: AFAIK, we are never going to parse this model, so it should be a regular interface, IMHO.
Zod is not very readable.

```suggestion
export interface Metadata {
  /**
   * Fields are used by search engines to index your site.
   */
  searchEngine: {
    /**
     * The ideal length for a title is approx. 60 characters.
     * When exceeding this limit, search engines will truncate the title.
     */
    title: string;

    /**
     * The ideal length is between 70 and 155 characters.
     * Minium length is 70 characters.
     * When exceeding this limit, search engines will truncate the description.
     */
    description: string;
  };

  /**
   * Fields are used by social media platforms to display your site.
   */
  socialMedia: {
    /**
     * The ideal length for a title is approx. 47 characters.
     * When exceeding this limit, platforms will truncate the title.
     *
     * LinkedIn will truncate the title at 119 characters.
     */
    title: string;

    /**
     * Should not exceed 85 characters to fit most platforms.
     */
    description: string;

    /**
     * The ideal size is 1200 x 630 pixels.
     * When you are providing more than one image, the first image will be used, but some
     * platforms will allow the user to choose which image to use.
     */
    images: Picture[];
  };
}
```

**Pull request closed without merging**